### PR TITLE
Huseyin- Week4 JS(added homework week4)

### PR DESCRIPTION
**Pull request by stphncrt**

### CODE REVIEW

File: 1-JavaScript/Week3/homework/ex1-giveCompliment.js [hunk 1]
```diff
@@ -18,27 +18,30 @@ Full description at: https://github.com/HackYourFuture/Homework/tree/main/1-Java
    Use `console.log` each time to display the return value of the
    `giveCompliment` function to the console.
 -----------------------------------------------------------------------------*/
-function giveCompliment(/* TODO parameter(s) go here */) {
-  // TODO complete this function
+function giveCompliment(name) {
+
+    const compliments = ['great', 'awesome', 'perfect', 'good', 'unique', 'handsome', 'excellent', 'gorgeous', 'beautiful', 'outstanding'];
+    const randomNumber = Math.floor(Math.random() * 10);
```
Comment (killoncio) [inline 1-JavaScript/Week3/homework/ex1-giveCompliment.js:24]: Why `*10` ?

File: 1-JavaScript/Week3/homework/ex4-shoppingCart.js [hunk 2]
```diff
@@ -23,54 +23,62 @@ you have more than 3 items in your shopping cart the first item gets taken out.
...
-function addToShoppingCart(/* parameters go here */) {
-  // TODO complete this function
+function addToShoppingCart(item) {
+    if (item) {
+        shoppingCart.push(item);
+        shoppingCart.length > 3 ? shoppingCart.shift() : null;
+        return `You bought ${shoppingCart.join(', ')}!`;
+    } else {
+
+        return `You bought ${shoppingCart.join(', ')}!`;
```
Comment (killoncio) [inline 1-JavaScript/Week3/homework/ex4-shoppingCart.js:33]: I added comment about this time ago in the Slack chat

File: 1-JavaScript/Week4/homework/ex1-doubleEvenNumbers.test.js [hunk 3]
```diff
@@ -11,19 +11,16 @@ Let's rewrite it (or _refactor_ it, as experienced developers would call it):
...
-  // TODO rewrite the function body using `map` and `filter`.
-  const newNumbers = [];
-  for (let i = 0; i < numbers.length; i++) {
-    if (numbers[i] % 2 === 0) {
-      newNumbers.push(numbers[i] * 2);
-    }
-  }
-  return newNumbers;
+
+    let newNumbers = [];
```
Comment (killoncio) [inline 1-JavaScript/Week4/homework/ex1-doubleEvenNumbers.test.js:15]: Why do you create first `newNumbers` variable, and then assign the value 2 rows later?

File: 1-JavaScript/Week4/homework/ex1-doubleEvenNumbers.test.js [hunk 4]
```diff
@@ -11,19 +11,16 @@ Let's rewrite it (or _refactor_ it, as experienced developers would call it):
...
+    const filteredNumbers = numbers.filter((number) => number % 2 === 0);
```
Comment (killoncio) [inline 1-JavaScript/Week4/homework/ex1-doubleEvenNumbers.test.js:16]: Can you think of a more descriptive name than `filteredNumbers` ? What is that variable storing? That should give you a clue.

Comment (killoncio) [reply 1-JavaScript/Week4/homework/ex1-doubleEvenNumbers.test.js:16]: Same for `newNumbers`

File: 1-JavaScript/Week4/homework/ex2-mondaysWorth.test.js [hunk 5]
```diff
@@ -10,42 +10,44 @@ Full description atL https://github.com/HackYourFuture/Homework/tree/main/1-Java
...
+        name: 'Talk to different members from the product team',
+        duration: 60,
+    },
 ];
 
 const hourlyRate = 25;
 
-function computeEarnings(/* TODO parameter(s) go here */) {
-  // TODO complete this function
+function computeEarnings(tasks, rate) {
```
Comment (killoncio) [inline 1-JavaScript/Week4/homework/ex2-mondaysWorth.test.js:33]: Well done!

File: 1-JavaScript/Week4/homework/ex3-lemonAllergy.test.js [hunk 6]
```diff
@@ -16,36 +16,39 @@ Complete the function called `sanitizeFruitBasket`:
...
+    'lemon',
+    'banana',
+    'watermelon',
+    'lemon',
 ];
 
 // ! Function under test
-function sanitizeFruitBasket(/* TODO parameter(s) go here */) {
-  // TODO complete this function
+function sanitizeFruitBasket(basket, fruit) {
```
Comment (killoncio) [inline 1-JavaScript/Week4/homework/ex3-lemonAllergy.test.js:29]: Nice!

File: 1-JavaScript/Week4/homework/ex3-lemonAllergy.test.js [hunk 7]
```diff
@@ -16,36 +16,39 @@ Complete the function called `sanitizeFruitBasket`:
...
-    // TODO replace next line with your code
-    expect(false).toBe(true);
-  });
-});
+    test('should take two parameters', () => {
+        expect(sanitizeFruitBasket).toHaveLength(2);
+
+    });
+
+    test('should not modify the original `fruitBasket` array', () => {
```
Comment (killoncio) [inline 1-JavaScript/Week4/homework/ex3-lemonAllergy.test.js:42]: If I understand correctly, in your test you are doing:
1) Clone an array
2) Testing that original and cloned array are the same.

I think you are missing a step :)

File: 1-JavaScript/Week4/homework/ex3-lemonAllergy.test.js [hunk 8]
```diff
@@ -16,36 +16,39 @@ Complete the function called `sanitizeFruitBasket`:
...
+        // Save the original contents of the fruit basket
+        const originalFruitBasketContents = [...fruitBasket];
+        expect(originalFruitBasketContents).toEqual(fruitBasket);
+
+    });
+
+    test('should return a new array that does not include the unwanted `lemon`', () => {
+        const actual = sanitizeFruitBasket(fruitBasket, 'lemon')
+        const fruitList = ["apple", "grapefruit", "banana", "watermelon"];
```
Comment (killoncio) [inline 1-JavaScript/Week4/homework/ex3-lemonAllergy.test.js:51]: Can you think of a way where you don't need to create an `expectedBasket` array?

File: 1-JavaScript/Week4/homework/ex4-observable/ex4-observable.js [hunk 9]
```diff
@@ -14,16 +14,16 @@ Complete the `createObservable()` function as follows:
...
-  const subscribers = [];
-  return {
-    subscribe: function (subscriber) {
-      // TODO complete this function
-    },
-    notify: function (message) {
-      // TODO complete this function
-    },
-  };
+    const subscribers = [];
```
Comment (killoncio) [inline 1-JavaScript/Week4/homework/ex4-observable/ex4-observable.js:17]: Well done!

File: 1-JavaScript/Week4/homework/ex1-doubleEvenNumbers.test.js (hunk 4 above)
Comment (stphncrt) [reply 1-JavaScript/Week4/homework/ex1-doubleEvenNumbers.test.js:16]: Fixed

File: 1-JavaScript/Week4/homework/ex1-doubleEvenNumbers.test.js (hunk 3 above)
Comment (stphncrt) [reply 1-JavaScript/Week4/homework/ex1-doubleEvenNumbers.test.js:15]: Fixed.

File: 1-JavaScript/Week4/homework/ex3-lemonAllergy.test.js (hunk 7 above)
Comment (stphncrt) [reply 1-JavaScript/Week4/homework/ex3-lemonAllergy.test.js:42]: Fixed

File: 1-JavaScript/Week4/homework/ex3-lemonAllergy.test.js (hunk 8 above)
Comment (stphncrt) [reply 1-JavaScript/Week4/homework/ex3-lemonAllergy.test.js:51]: Fixed

File: 1-JavaScript/Week4/homework/ex4-observable/ex4-observable.js (hunk 9 above)
Comment (stphncrt) [reply 1-JavaScript/Week4/homework/ex4-observable/ex4-observable.js:17]: Thanks for your all precious comments.

**Comment by killoncio** [comment]
Will close this PR since it's a duplicate of https://github.com/HackYourHomework/Homework-class35/pull/47/files , will follow up there.

**Pull request closed without merging**